pkg/runner/runner.go

```diff
@@ -137,48 +137,33 @@ func Run(cfg *Config) error {
 		}
 	}
 
-	var (
-		group       run.Group
-		reloadRules func() error
-	)
-	pendingTelemetry := map[string]string{
-		"rules_enabled":    "false", // Will be written in telemetry table as `promscale_rules_enabled: false`
-		"alerting_enabled": "false", // `promscale_alerting_enabled: false`
-	}
-	if cfg.RulesCfg.ContainsRules() {
-		pendingTelemetry["rules_enabled"] = "true"
-		if cfg.RulesCfg.ContainsAlertingConfig() {
-			pendingTelemetry["alerting_enabled"] = "true"
-		} else {
-			log.Debug("msg", "Alerting configuration not present in the given Prometheus configuration file. Alerting will not be initialized")
-		}
-
-		rulesCtx, stopRuler := context.WithCancel(context.Background())
-		defer stopRuler()
-
-		var manager *rules.Manager
-		manager, reloadRules, err = rules.NewManager(rulesCtx, prometheus.DefaultRegisterer, client, &cfg.RulesCfg)
-		if err != nil {
-			return fmt.Errorf("error creating rules manager: %w", err)
-		}
-		cfg.APICfg.Rules = manager
-
-		if err := reloadRules(); err != nil {
-			return err
-		}
+	telemetryEngine := initTelemetryEngine(client)
+	telemetryEngine.Start()
+	defer telemetryEngine.Stop()
 
-		group.Add(
-			func() error {
-				log.Info("msg", "Started Rule-Manager")
-				return manager.Run()
-			}, func(error) {
-				log.Info("msg", "Stopping Rule-Manager")
-				stopRuler()
-			},
-		)
-	} else {
-		log.Debug("msg", "Rules files not found in the given Prometheus configuration file. Both rule-manager and alerting will not be initialized")
+	rulesCtx, stopRuler := context.WithCancel(context.Background())
+	defer stopRuler()
+	manager, reloadRules, err := rules.NewManager(rulesCtx, prometheus.DefaultRegisterer, client, &cfg.RulesCfg, telemetryEngine)
+	if err != nil {
+		return fmt.Errorf("error creating rules manager: %w", err)
 	}
+	cfg.APICfg.Rules = manager
+
+	var group run.Group
+	group.Add(
+		func() error {
+			// Reload the rules before starting the rules-manager to ensure all rules are healthy.
```

Comment:
I'd actually not bound these two together. If reloading of rules fail we should not just stop rule engine but rather stop Promscale Otherwise things can just stop and user not being aware.